src/handlers/ripgrep_json.rs

```diff
@@ -15,7 +15,16 @@ pub fn parse_line(line: &str) -> Option<grep::GrepLine> {
             // A real line of rg --json output, i.e. either of type "match" or
             // "context".
             let mut code = ripgrep_line.data.lines.text;
-            if code.ends_with('\n') {
+            // ripgrep --json emits a trailing newline for each match, but
+            // delta's code highlighting does not expect a trailing newline.
+            // Therefore, remove the trailing newline if it is not part of the
+            // match.
```

Comment:
If delta needs to strip the `\n` in the usual case, then that must be because some bad thing X happens if it does not do it. So why is it OK to not strip the `\n` when it is part of the match? Why does X not happen?

Reply:
Because once it will be printed, the \n will be stripped since it's included in the match, unless im missing something here

Reply:
I'm sorry, I'm not following. Would you mind explaining in more detail? I'm not understanding the relationship between whether or not something is in the match, and the stripping newlines for the highlighter. I think it's me that's missing something!

Reply:
I need some time, I've been trying with the example you mentioned in the #1496  that's failing, i should look more into it, specially that now i am thinking about it i don't think i handled it well it's a bit confusing